File: services/frontend/src/types/Pedido.ts
/**
 * Pedido Types
 * 
 * Baseado no schema.sql e endpoints do Catalog Service
 * Pedidos de renovação ou revogação de acesso
 */

/**
 * Pedido (completo)
 */
export interface Pedido {
  id: string;
  cliente_id: string;
  estado: "pendente" | "aprovado" | "rejeitado";
  tipo_pedido: "renovação" | "revogação";
  criado_em: string;        // ISO datetime
  gerido_por: string | null; // ID do admin que aprovou/rejeitou
}

/**
 * Pedido Create (POST /api/pedidos/)
 */
export interface PedidoCreate {
  cliente_id: string;
  tipo_pedido: "renovação" | "revogação";
}

/**
 * Pedido Response (alias para Pedido)
 */
export type PedidoResponse = Pedido;

/**
 * Pedido com informações do cliente (para listagem)
 */
export interface PedidoComCliente extends Pedido {
  cliente_nome: string;
  cliente_email: string;
  data_expiracao_atual: string;
}

/**
 * Request para aprovar pedido
 */
export interface ApprovalRequest {
  admin_id: string;
  nova_data_expiracao?: string;  // Opcional, usado em renovações
}

/**
 * Request para rejeitar pedido
 */
export interface RejectionRequest {
  admin_id: string;
}

File: services/catalog/app/models/pedido.py
"""
Pedido Models
Models Pydantic para Pedidos de renovação/revogação
"""
from pydantic import BaseModel, Field
from typing import Optional


class PedidoBase(BaseModel):
    """Base model para Pedido"""
    cliente_id: str = Field(..., description="ID do cliente")
    tipo_pedido: str = Field(..., pattern="^(renovação|revogação)$")


class PedidoCreate(PedidoBase):
    """Model para criar Pedido"""
    pass


class PedidoUpdate(BaseModel):
    """Model para atualizar Pedido (aprovar/rejeitar)"""
    estado: str = Field(..., pattern="^(pendente|aprovado|rejeitado)$")
    gerido_por: str = Field(..., description="ID do admin que aprovou/rejeitou")
    nova_data_expiracao: Optional[str] = Field(None, description="Nova data se aprovado")


class PedidoResponse(PedidoBase):
    """Model para resposta de Pedido"""
    id: str
    estado: str
    criado_em: str
    gerido_por: Optional[str]
    
    class Config:
        from_attributes = True

class PedidoResponseCliente(PedidoResponse):
    cliente_nome: str
    cliente_email:str
    data_expiracao_atual: str


class PedidoApprove(BaseModel):
    """Model para aprovar pedido"""
    admin_id: str = Field(..., description="ID do admin aprovando")

class PedidoReject(BaseModel):
    """Model para rejeitar pedido"""
    admin_id: str = Field(..., description="ID do admin rejeitando")
